recipes/rstudio/build-desktop.sh

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env bash
+
+## These variables have to be set correctly.
+## RSTUDIO_VERSION_SUFFIX will need to be updated here for every new version bump
+export RSTUDIO_VERSION_MAJOR=$(echo ${PKG_VERSION} | cut -d. -f1)
+export RSTUDIO_VERSION_MINOR=$(echo ${PKG_VERSION} | cut -d. -f2)
+export RSTUDIO_VERSION_PATCH=$(echo ${PKG_VERSION} | cut -d. -f3)
+export RSTUDIO_VERSION_SUFFIX="+443"
+export GIT_COMMIT=fc9e217
+export PACKAGE_OS=$(uname -om)
+
+export BUILD_TYPE=Release
+export RSTUDIO_TARGET=Desktop
+
+export RSTUDIO_TOOLS_ROOT="${PREFIX}/opt/rstudio-tools/$(uname -m)"
+mkdir -p $RSTUDIO_TOOLS_ROOT
+export SOCI_LIBRARY_DIR=${PREFIX}/lib
+export SOCI_DIR=$RSTUDIO_TOOLS_ROOT/soci
+export SOCI_BIN_DIR=$SOCI_DIR/build
+mkdir -p $SOCI_BIN_DIR
+ln -s ${PREFIX}/* ${SOCI_BIN_DIR}
+export SOCI_CORE_LIB=${PREFIX}/lib/libsoci_core.so
+export SOCI_POSTGRESQL_LIB=${PREFIX}/lib/libsoci_postgresql.so
+export SOCI_SQLITE_LIB=${PREFIX}/lib/libsoci_sqlite3.so
+
+## Instead of installing dependencies as instructed by the upstream
+## build documentation we create symlinks in the expected locations
+## to the conda-forge equivalents
+pushd dependencies/common
+_pandocver=$(grep -oP "(?<=PANDOC_VERSION=\").*(?=\"$)" install-pandoc)
+install -d pandoc/${_pandocver}
+ln -sfT ${PREFIX}/bin/pandoc pandoc/${_pandocver}/pandoc
+_nodever=$(grep -oP "(?<=NODE_VERSION=\").*(?=\"$)" ../tools/rstudio-tools.sh)
+install -d node
+ln -sfT ${BUILD_PREFIX} node/${_nodever}
+ln -sfT ${PREFIX}/share/hunspell_dictionaries dictionaries
+ln -sfT ${PREFIX}/lib/mathjax mathjax-27
+popd
+
+# Fix links for src/cpp/session/CMakeLists.txt
+pushd dependencies
+ln -sfT common/dictionaries dictionaries
+ln -sfT common/mathjax-27 mathjax-27
+ln -sfT common/pandoc pandoc
+ln -sfT common/node node
+popd
+
+cmake -S . -B build \
+      -DRSTUDIO_TARGET=Desktop \
+      -DCMAKE_BUILD_TYPE=Release \
+      -DCMAKE_INSTALL_PREFIX="${PREFIX}/lib/rstudio" \
```

Comment:
Patch https://github.com/rstudio/rstudio/blob/edd144e6c64f765f62716e57a31808597e92ab7e/src/cpp/session/include/session/SessionConstants.hpp#L143 to be just `bin` instead of `bin/pandoc`?

Reply:
Yeah, that will be nicer. OK, will do that instead, thanks!

Reply:
I have a local build that installs to `RPREFIX` instead of `PREFIX/lib/rstudio`, but now I'm not so sure this is a good idea. There is a lot of stuff in the top-level directory that doesn't belong there, e.g., `COPYING`, `INSTALL` `www`. It looks to me like the build is designed to install rstudio to a stand-alone directory as I did before. Thoughts?

Reply:
defaults package also uses `PREFIX`, so it's better to use the same layout.

Reply:
OK I'll clean it up as best I can.

Reply:
Well I spent some more time on it and managed to break it quite nicely :-(

At this point I'm against trying to force it into a standard directory layout. The old defaults channel package managed this, but only with extensive patches (e.g., https://github.com/AnacondaRecipes/aggregateR/blob/be32d88d1ccc5735a34522f21f0e56dbd0849668/rstudio-feedstock/recipe/0028-Conda-Put-resources-in-share-rstudio.patch ; all together there are 45 patches needed for that recipe, and the maintenance burden of all those patches is  I think the main reason why we haven't had a working rstudio conda package for years now.

I'm happy to maintain a straight-up packing of the upstream releases, but maintaining an extensive set of local patches is more than I'm willing to do. Happy to get others thoughts on this too @conda-forge/help-r

Reply:
> Well I spent some more time on it and managed to break it quite nicely :-(

What broke?

> please let me know if you don't think that is acceptable.

Without knowing how things broke I can't say if this is acceptable or not.

Reply:
The build completes without error, but the resulting binary does not work because it looks for resources in the wrong place. My assessment is that the only way to get a freedesktop-like layout is to extensively patch the upstream sources, which I am unwilling to do.

Reply:
You can see the problem in https://github.com/conda-forge/staged-recipes/pull/18401

Reply:
When I try to run the binary produced by https://github.com/conda-forge/staged-recipes/pull/18401 I get this:
```
TTY detected. Printing informational message about logging configuration. Logging configuration loaded from '/home/izahn/.config/kdedefaults:/etc/xdg/rstudio/logging.conf'. Logging to '/home/izahn/.local/share/rstudio/log/rdesktop.log'.
2022-03-19T14:53:08.987141Z [rdesktop] ERROR system error 2 (No such file or directory) [path: /home/izahn/mambaforge/envs/rstudio/bin/resources/html/error.html]; OCCURRED AT rstudio::core::Error rstudio::core::FilePath::openForRead(std::shared_ptr<std::basic_istream<char> >&) const src/cpp/shared_core/FilePath.cpp:1461; LOGGED FROM: void rstudio::desktop::SessionLauncher::showLaunchErrorPage() src/cpp/desktop/DesktopSessionLauncher.cpp:379
"Uncaught TypeError: Cannot read property 'notifyRCrashed' of undefined"
2022-03-19T14:53:09.051025Z [rdesktop] ERROR system error 2 (No such file or directory) [path: /home/izahn/mambaforge/envs/rstudio/bin/resources/html/loading.html]; OCCURRED AT rstudio::core::Error rstudio::core::FilePath::openForRead(std::shared_ptr<std::basic_istream<char> >&) const src/cpp/shared_core/FilePath.cpp:1461; LOGGED FROM: void rstudio::desktop::MainWindow::onLoadFinished(bool) src/cpp/desktop/DesktopMainWindow.cpp:714
2022-03-19T14:53:09.069802Z [rdesktop] ERROR system error 2 (No such file or directory) [path: /home/izahn/mambaforge/envs/rstudio/bin/resources/html/loading.html]; OCCURRED AT rstudio::core::Error rstudio::core::FilePath::openForRead(std::shared_ptr<std::basic_istream<char> >&) const src/cpp/shared_core/FilePath.cpp:1461; LOGGED FROM: void rstudio::desktop::MainWindow::onLoadFinished(bool) src/cpp/desktop/DesktopMainWindow.cpp:714
2022-03-19T14:53:09.304886Z [rdesktop] ERROR system error 2 (No such file or directory) [path: /home/izahn/mambaforge/envs/rstudio/bin/resources/html/loading.html]; OCCURRED AT rstudio::core::Error rstudio::core::FilePath::openForRead(std::shared_ptr<std::basic_istream<char> >&) const src/cpp/shared_core/FilePath.cpp:1461; LOGGED FROM: void rstudio::desktop::MainWindow::onLoadFinished(bool) src/cpp/desktop/DesktopMainWindow.cpp:714
2022-03-19T14:53:09.724277Z [rdesktop] ERROR system error 2 (No such file or directory) [path: /home/izahn/mambaforge/envs/rstudio/bin/resources/html/loading.html]; OCCURRED AT rstudio::core::Error rstudio::core::FilePath::openForRead(std::shared_ptr<std::basic_istream<char> >&) const src/cpp/shared_core/FilePath.cpp:1461; LOGGED FROM: void rstudio::desktop::MainWindow::onLoadFinished(bool) src/cpp/desktop/DesktopMainWindow.cpp:714
2022-03-19T14:53:10.344989Z [rdesktop] ERROR system error 2 (No such file or directory) [path: /home/izahn/mambaforge/envs/rstudio/bin/resources/html/loading.html]; OCCURRED AT rstudio::core::Error rstudio::core::FilePath::openForRead(std::shared_ptr<std::basic_istream<char> >&) const src/cpp/shared_core/FilePath.cpp:1461; LOGGED FROM: void rstudio::desktop::MainWindow::onLoadFinished(bool) src/cpp/desktop/DesktopMainWindow.cpp:714
2022-03-19T14:53:11.163897Z [rdesktop] ERROR system error 2 (No such file or directory) [path: /home/izahn/mambaforge/envs/rstudio/bin/resources/html/loading.html]; OCCURRED AT rstudio::core::Error rstudio::core::FilePath::openForRead(std::shared_ptr<std::basic_istream<char> >&) const src/cpp/shared_core/FilePath.cpp:1461; LOGGED FROM: void rstudio::desktop::MainWindow::onLoadFinished(bool) src/cpp/desktop/DesktopMainWindow.cpp:714
2022-03-19T14:53:12.183333Z [rdesktop] ERROR system error 2 (No such file or directory) [path: /home/izahn/mambaforge/envs/rstudio/bin/resources/html/loading.html]; OCCURRED AT rstudio::core::Error rstudio::core::FilePath::openForRead(std::shared_ptr<std::basic_istream<char> >&) const src/cpp/shared_core/FilePath.cpp:1461; LOGGED FROM: void rstudio::desktop::MainWindow::onLoadFinished(bool) src/cpp/desktop/DesktopMainWindow.cpp:714
2022-03-19T14:53:13.403856Z [rdesktop] ERROR system error 2 (No such file or directory) [path: /home/izahn/mambaforge/envs/rstudio/bin/resources/html/loading.html]; OCCURRED AT rstudio::core::Error rstudio::core::FilePath::openForRead(std::shared_ptr<std::basic_istream<char> >&) const src/cpp/shared_core/FilePath.cpp:1461; LOGGED FROM: void rstudio::desktop::MainWindow::onLoadFinished(bool) src/cpp/desktop/DesktopMainWindow.cpp:714
2022-03-19T14:53:14.827088Z [rdesktop] ERROR system error 2 (No such file or directory) [path: /home/izahn/mambaforge/envs/rstudio/bin/resources/html/loading.html]; OCCURRED AT rstudio::core::Error rstudio::core::FilePath::openForRead(std::shared_ptr<std::basic_istream<char> >&) const src/cpp/shared_core/FilePath.cpp:1461; LOGGED FROM: void rstudio::desktop::MainWindow::onLoadFinished(bool) src/cpp/desktop/DesktopMainWindow.cpp:714
```
I uploaded it to my izahn channel to make it easier to install and test locally if you want.

Reply:
We spent some time on this in https://github.com/conda-forge/staged-recipes/pull/18401 without success. I also have concerns about the patches there causing other problems. It's this is not broken I'm not going to fix it. #wontfix